terraform/terraform_datadog_monitors.tf

```diff
@@ -0,0 +1,172 @@
+      version = "3.10.0"
+    }
+  }
+}
+
+#Monitor threshold values
+variable "time_to_contact_issues_warning" {
+  type    = number
+  default = 5
+}
+variable "time_to_contact_issues_critical" {
+  type    = number
+  default = 30
+}
+variable "time_to_contact_PRs_warning" {
+  type    = number
+  default = 5
+}
+variable "time_to_contact_PRs_critical" {
+  type    = number
+  default = 10
+}
+variable "time_to_resolve_issue_warning" {
+  type    = number
+  default = 30
+}
+variable "time_to_resolve_issue_critical" {
+  type    = number
+  default = 180
+}
+variable "open_issue_count_warning" {
+  type    = number
+  default = 50
+}
+variable "open_issue_count_twilio_critical" {
+  type    = number
+  default = 84
+}
+variable "open_issue_count_sendgrid_critical" {
+  type    = number
+  default = 80
+}
+
+#Twilio Monitors
+resource "datadog_monitor" "time_to_contact_issues_twilio" {
+  name    = "Mean time to contact for Issues for Twilio Org"
+  type    = "metric alert"
+  message = "@slack-Twilio-alerts-dev-interfaces ```{{#is_alert}} ALERT! Mean time to contact for Issues for Twilio org= {{value}}. We are in breach of our SLO (<=30 days). {{override_priority 'P1'}} {{/is_alert}} {{#is_warning}} WARNING! Mean time to contact for Issues for Twilio org= {{value}}. Our SLA threshold is <= 30 days. {{override_priority 'P3'}} {{/is_warning}} ```"
+  query   = "sum(last_5m):max:library.time_to_contact.mean{org:twilio} >= 30"
+
+  monitor_thresholds {
+    warning  = var.time_to_contact_issues_warning
+    critical = var.time_to_contact_issues_critical
+  }
+
+  tags = ["owner:developer_experience", "team:dev-Interfaces"]
+}
+
+resource "datadog_monitor" "time_to_contact_PRs_twilio" {
+  name    = "Mean time to contact for PRs for Twilio Org"
+  type    = "metric alert"
+  message = "@slack-Twilio-alerts-dev-interfaces ```{{#is_alert}} ALERT! Mean time to contact for PRs for Twilio org= {{value}}. We are in breach of our SLO (<=10 days). {{override_priority 'P1'}} {{/is_alert}} {{#is_warning}} WARNING! Mean time to contact for PRs for Twilio org= {{value}}. Our SLA threshold is <= 10 days. {{override_priority 'P3'}} {{/is_warning}} ```"
+  query   = "sum(last_5m):max:library.time_to_contact_pr.mean{org:twilio} >= 10"
+
+  monitor_thresholds {
+    warning  = var.time_to_contact_PRs_warning
+    critical = var.time_to_contact_PRs_critical
+  }
+
+  tags = ["owner:developer_experience", "team:dev-Interfaces"]
+}
+
+resource "datadog_monitor" "time_to_resolve_issue_twilio" {
+  name    = "Time to resolve issues for Twilio Org"
+  type    = "metric alert"
+  message = "@slack-Twilio-alerts-dev-interfaces ```{{#is_alert}} ALERT! Mean time to contact for Issues for SendGrid org= {{value}}. We are in breach of our SLO (<=30 days). {{override_priority 'P1'}} {{/is_alert}} {{#is_warning}} WARNING! Mean time to contact for Issues for SendGrid org= {{value}}. Our SLA threshold is <= 30 days. {{override_priority 'P3'}} {{/is_warning}} ```"
```

Comment:
Should be referencing the `time_to_resolve_issue_warning` and `time_to_resolve_issue_critical` vars in this message. Otherwise they're defined multiple times and will get out of sync. Same for the query value.

Reply:
Fixed!